### PR TITLE
Fix erroneous incident count on rapid table page change

### LINKED ISSUE #339: Bug: the pages goes into the negative
**Describe the bug**
If the user click on the "previous page" arrow fast enough you can go way past "page 1" all the way to -1000 if the user clicks fast enough.

**To Reproduce**
1. Click on the next page arrow on the bottom right 2 times
2. Click rapidly on the "previous page"-icon and you will see that you can go all the way to the negative side.

**Expected behavior**
Expected to stop at the first page

**Screenshots**
argus

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version 95

### PR DESCRIPTION
**Pull request by podliashanyk**
This is done by making navigation arrows disabled both while whole table ir reloading, **and** while _page change_ event is loading.

Closes #339

### CODE REVIEW

**Review by stveit — CHANGES_REQUESTED** [review]
Seems like a decent idea for a fix, page buttons are disabled while the list is loading, but are back as soon as the page has loaded.
A downside to this is that you cant really go down multiple pages fast, so it might take a while to go deep, but that kinda makes sense as the website doesnt actually know how deep you can go without paging all the way, so its more a limitation of the paging in the backend than anything else

But page count is still messed up a bit. Example: with pageocunt 50 and a total of 165 incidents, on the last page it will show 151-165. If you go backwards, while its loading the count will show 101-115, when it should show 101-150.

Similar going forwards from 101-150: it will show 151-200, even though the total amount of incidents is 165. because of pagination, the frontend doesnt necessarily know this so it guesses at 200.

For the backwards problem, the frontend should know the correct amount of incidents, but for going forward, it might be a solution to update the incident numbering after the new page is loaded and you know what the true count should be (first naive idea i had for a solution, no need to take this as gospel)

File: src/components/incidenttable/FilteredIncidentTable.tsx [hunk 1]
```diff
@@ -189,7 +193,7 @@ const FilteredIncidentTable = () => {
       setVirtCursor(DEFAULT_VIRT_CURSOR);
     };
 
-    const disabled = isLoading || false;
+    const disabled = isLoading || isCursorLoading || false;
```
[the remark this answers is not in the document]
Comment (podliashanyk) [reply src/components/incidenttable/FilteredIncidentTable.tsx:196]: Sounds right, I will double check and remove it if its redundant

**Comment by podliashanyk** [comment]
> But page count is still messed up a bit. Example: with pageocunt 50 and a total of 165 incidents, on the last page it will show 151-165. If you go backwards, while its loading the count will show 101-115, when it should show 101-150.
> 
> Similar going forwards from 101-150: it will show 151-200, even though the total amount of incidents is 165. because of pagination, the frontend doesnt necessarily know this so it guesses at 200.
> 
> For the backwards problem, the frontend should know the correct amount of incidents, but for going forward, it might be a solution to update the incident numbering after the new page is loaded and you know what the true count should be (first naive idea i had for a solution, no need to take this as gospel)

I have digged into the problem, and see that update of the _from_ and _to_ values of the pagination count is something that MUI TablePagination does for us. We do not even send any _from_ and _to_ values to the MUI TablePagination component. 

I have so far solved the problem with the misleading values (while the page is loading) by:
- displaying the MUI Skeleton instead of the actual values **while the page is loading**
- switching to displaying the actual values **after the page has loaded** and MUI TablePagination is aware of the "final" _from_, _to_ and _count_ values.

It seems possible to interfere with TablePagination's normal workflow and update the _from_ and _to_ in our code, but it is somewhat tedious (plus those values are actually displayed for like a second). **So the question is: is the solution with the MUI Skeleton good enough, or should I go for interfering with TablePagination?**

**Comment by stveit** [comment]
I think the Skeleton is OK. Perfect would be to show the actual true values, but we cant know the upper limit because pagination, so theres not that much reason to show anything (and as you say, it doesnt display for very long). I think this is fine, and we can potentially create a new issue for perfecting it and making it better later.

**Comment by hmpf** [comment]
Keep in mind that we very deliberately chose the type of pagination that we have because it will still work with an absurd amount of incidents. We could swap to actual paged pagination, not cursor, and move the problem to the database instead (NO THANKS).

**Review by stveit — APPROVED** [review]
I think this is fine so I'll approve. As @hmpf says, the limitations are consequences of the pagination method.

**Pull request merged**